packages/core/src/snark/snark-child-process.ts

```diff
@@ -8,11 +8,12 @@ process.on(
     const { vk, proof, signals } = message
     let result!: boolean
     try {
-      result = snarkjs.groth16.verify(
+      snarkjs.groth16.verify(
```

Comment:
I think this change is breaking. The `verify` function should return a boolean and throw an error if it's malformed. I might be wrong but I feel like I tried something similar.

Reply:
https://github.com/iden3/snarkjs/blob/v0.3.33/src/groth16_verify.js
snarkjs' `verify` function was returning a `Promise` object and I think we should change the code like
```typescript
    let result!: boolean
    try {
      result = await snarkjs.groth16.verify(
        ffjs.utils.unstringifyBigInts(vk),
        ffjs.utils.unstringifyBigInts(signals),
        ffjs.utils.unstringifyBigInts(proof),
      )
    } catch (e) {
      logger.error(e)
      result = false
    }
```

Reply:
Please see https://github.com/zkopru-network/zkopru/pull/257/commits/50c03e631f1148abde81d1928606673fff88a8db

Reply:
Ah yes, I was returning the promise from the async function, but it's more clear to `await` it.